### PR TITLE
feat(framework): support MegEngine model

### PR DESCRIPTION
**Pull request by FateScript**
This PR supports a new deep learning framework named [MegEngine](https://github.com/MegEngine/MegEngine). 
Please kindly review.
Related issue #451

### CODE REVIEW

**Comment by bmartinn** [comment]
Thanks @FateScript , this looks great.
Can you please also provide a toy example for testing and demonstration ?
(something to put under `examples/frameworks/megengine/`)

**Comment by FateScript** [comment]
Updated now. Please review kindly, thanks @bmartinn

**Comment by bmartinn** [comment]
Thanks @FateScript , looks good.
Do you want to add some metric logging? is there a standard way to report metrics with `meghengine` ? Should we just use the direct `clearml.Logger` class, wdyt?

**Comment by FateScript** [comment]
> Thanks @FateScript , looks good.
> Do you want to add some metric logging? is there a standard way to report metrics with `meghengine` ? Should we just use the direct `clearml.Logger` class, wdyt?

For most users of megengine, `tensorboard` or `megengine.logger` is the standard way to report metrics. I will open a PR if any user wants to report metrics with `clearml.Logger` in the future.

BTW, after talk with `megengine` team, `pkl` is the only official recommanded way to save model state_dict. I push a new commit to fix this. Please kindly review, thanks.

**Comment by FateScript** [comment]
gently ping @bmartinn

**Comment by bmartinn** [comment]
Thanks @FateScript, it looks good to me. 
Let me make sure the test passes, and we could merge (it probably will take a few days)
